src/index.d.ts

```diff
@@ -0,0 +1,52 @@
+import { Transform } from "readable-stream";
+
+export declare module 'encoding' {
+    export function convert(buf: Buffer, toCharset: string, fromCharset: string): Buffer;
+}
+
+export interface Compiler {
+    _table: GetTextTranslations;
+    compile(): Buffer;
+}
+
+export interface GetTextComment {
+    translator?: string;
+    reference?: string;
+    extracted?: string;
+    flag?: string;
+    previous?: string;
+}
+
+export interface GetTextTranslation {
+    msgctxt?: string;
+    msgid: string;
+    msgid_plural?: string;
+    msgstr: string[];
+    comments?: GetTextComment;
+}
+
+export interface GetTextTranslations {
+    charset: string;
+    headers: { [headerName: string]: string };
+    translations: { [msgctxt: string]: { [msgId: string]: GetTextTranslation } };
+}
+
+export interface parserOptions {
+    defaultCharset?: string;
+    validation?: boolean;
+}
+
+export interface po {
+    parse: (buffer: Buffer | string, defaultCharset?: string) => GetTextTranslations;
+    compile: (table: GetTextTranslations, options?: parserOptions) => Buffer;
+    createParseStream: (options?: parserOptions, transformOptions?: import('readable-stream').TransformOptions) => Transform;
+}
+
+export interface mo {
+    parse: (buffer: Buffer | string, defaultCharset?: string) => GetTextTranslations;
+    compile: (table: GetTextTranslations, options?: parserOptions) => Buffer;
+}
+
+export * from "./@types";
```

Comment:
To me what is is not 100% correct of that, is to export the types related to the functions provided by gettext-parser in i.e. this file [this](https://github.com/smhg/gettext-parser/blob/42a969325c052b9b1927291b4d6a15172298e21f/src/types.d.ts) doing this https://github.com/smhg/gettext-parser/blob/42a969325c052b9b1927291b4d6a15172298e21f/src/types.d.ts#L37-L46

but this is required i guess until we decide to switch completely to typescript because the jsdoc didn't provide the required output. Furthermore these are exactly like the ones that are now on the [strictly typed repo](https://www.npmjs.com/package/@types/gettext-parser?activeTab=code), so, in order to avoid any possible glitches for users I had to take them and restore (approximately) as they were before

Reply:
fixed with https://github.com/smhg/gettext-parser/pull/89/commits/cbe6e8f05e6455ea79023b12af39d20077b6c1d2

Since having a `.d.ts` declarations file would have complicated things I moved the definitions into a separate `.js` file. In that way we get the same result because tsc will generates a nearly identical definitions file them during compilation